### DISCUSSION TITLE
Issue with Firefox driver

### DISCUSSION THREAD

**Discussion by gsumit1** [serenity-bdd/serenity-core#2713] [Q&A]
With webdriver.driver=firefox settings in Serenety.properties file for both the below projects
https://github.com/serenity-bdd/serenity-cucumber-starter
https://github.com/serenity-bdd/serenity-junit-starter

I am facing the gecko driver path issue for the windows machine. I have verified for 2 different machines to ensure that this issue is not due to the machines whereas it works for chrome & edge without any issues.

I am using <serenity.version>3.2.0</serenity.version> in POM

In fact, it's not copying the geckodriver in C:\Users\xxxx\.cache\selenium where the webDriverManager usually copied the driver.

![Firefox issue](https://user-images.githubusercontent.com/17493208/154689566-aea9dac3-1202-4219-92ef-99c56d528263.JPG)

**Comment by wakaleo** [comment]
Maybe a problem with WebDriverManager on your machine? The Gecko driver should be downloaded automatically by WebDriverManager - there is no special configuration to do. I generally don't use firefox because I find it unreliable, but when I do it seems to work just fine.

**Reply by gsumit1** [reply]
I agree with you, nowadays we use Firefox less. But this is an **issue** introduced on **3.1.0**.  I have tested it on both Mac & Windows machines, same issue.

It is working fine for 3.0.5.

![image](https://user-images.githubusercontent.com/17493208/154814781-7be52e6b-fcde-4957-9f1a-7fd5ab52b8e5.png)

Please have a look into this.

Thank You
Sumit

**Reply by wakaleo** [reply]
I don't recall anything changing in the Firefox code, but feel free to take a look at the code and propose a PR if you find anything. Otherwise, if it is important for your projects, you can always look into getting one of the commercial support packages.
